firestore/app/src/main/java/com/google/example/firestore/DocSnippets.java

```diff
@@ -128,18 +130,27 @@ public void setup() {
         // [END get_firestore_instance]
 
         // [START set_firestore_settings]
-        FirebaseFirestoreSettings settings = new FirebaseFirestoreSettings.Builder()
-                .setPersistenceEnabled(true)
-                .build();
-        db.setFirestoreSettings(settings);
+        FirebaseFirestoreSettings settings = 
+        new FirebaseFirestoreSettings.Builder(db.getFirestoreSettings())
+            // Use memory-only cache
+            .setLocalCacheSettings(MemoryCacheSettings.newBuilder().build())
+            // Use persistent disk cache (default)
+            .setLocalCacheSettings(PersistentCacheSettings.newBuilder()
+                                    .build())
+            .build();
```

Comment:
Shall we keep the call to `setFirestoreSettings()`?

```suggestion
        FirebaseFirestoreSettings settings = 
        new FirebaseFirestoreSettings.Builder(db.getFirestoreSettings())
            // Use memory-only cache
            .setLocalCacheSettings(MemoryCacheSettings.newBuilder().build())
            // Use persistent disk cache (default)
            .setLocalCacheSettings(PersistentCacheSettings.newBuilder()
                                    .build())
            .build();
        db.setFirestoreSettings(settings);
```